File: src/main/java/com/saybetter/domain/solution/ui/dto/SolutionResponse.java
package com.saybetter.domain.solution.ui.dto;

import java.util.List;

import lombok.Builder;

public class SolutionResponse {

	private SolutionResponse() {
		throw new IllegalStateException("Utility class");
	}

	@Builder
	public static class SymbolList {
		private String name;
		private List<SymbolInfo> symbols;
	}

	@Builder
	public static class SymbolInfo {
		private String description;
		private String imageUrl;
	}
}

File: core/core-api/src/main/kotlin/io/say/better/domain/solution/ui/dto/SolutionResponse.kt
package io.say.better.domain.solution.ui.dto

class SolutionResponse private constructor() {
    init {
        throw IllegalStateException("Utility class")
    }

    data class SymbolList(
        val name: String,
        val symbols: List<SymbolInfo>? = null,
    )

    data class SymbolInfo(
        val description: String,
        val imageUrl: String,
    )

    data class ProgressInfo(
        val progressId: Long,
    )
}
